GitUpKit/Interface/GIGraphView.m

```diff
@@ -860,112 +860,164 @@ static void _DrawLine(GILine* line, CGContextRef context, CGFloat offset, CGFloa
   }
 }
 
-static void _DrawBranchTitle(CGContextRef context, CGFloat x, CGFloat y, GIBranch* branch, NSDictionary* attributes, GIGraphOptions options) {
-  NSUInteger index;
+static void _DrawBranchTitle(CGContextRef context, CGFloat x, CGFloat y, NSColor* color, GIBranch* branch, GIGraphOptions options) {
+
+  // Cache common format for multiline string
+  static NSMutableDictionary* multilineTitleAttributes = nil;
+  if (multilineTitleAttributes == nil) {
+    multilineTitleAttributes = [[NSMutableDictionary alloc] init];
+    
+    CTFontRef titleFont = CTFontCreateUIFontForLanguage(kCTFontUIFontSystem, 13.0, CFSTR("en-US"));
+    multilineTitleAttributes[NSFontAttributeName] = (id)titleFont;
+    CFRelease(titleFont);
+    
+    NSMutableParagraphStyle* style = [[NSMutableParagraphStyle alloc] init];
+    style.lineHeightMultiple = 0.8;
+    multilineTitleAttributes[NSParagraphStyleAttributeName] = style;
+    [style release];
+  }
+  multilineTitleAttributes[NSForegroundColorAttributeName] = [color shadowWithLevel:0.25];
   
-  // Generate text
+  // Cache bold font and calculate darker color for building multiline string
+  static CTFontRef boldFont = NULL;
+  if (boldFont == NULL) {
+    boldFont = CTFontCreateUIFontForLanguage(kCTFontUIFontEmphasizedSystem, 13.0, CFSTR("en-US"));
+  }
+  NSColor* darkColor = [color shadowWithLevel:0.7];
+  
+  // Start new attributed string for the branch title
+  NSMutableAttributedString* multilineTitle = [[NSMutableAttributedString alloc] initWithString:@""];
+  [multilineTitle beginEditing];
   
-  NSMutableString* label = [[NSMutableString alloc] init];
-  index = 0;
   for (GCHistoryLocalBranch* localBranch in branch.localBranches) {
-    if (index) {
-      [label appendString:@", "];
-    }
-    NSString* upstream = localBranch.upstream.name;
-    if (upstream) {
-      [label appendFormat:@"%@ ◀ %@", localBranch.name, upstream];
+    
+    NSString* upstreamName = localBranch.upstream.name;
+    if (upstreamName) {
+      
+      NSString* localBranchName = localBranch.name;
+      NSRange localBranchNameRange = NSMakeRange(multilineTitle.length, localBranchName.length);
+      _AppendAttributedString((CFMutableAttributedStringRef)multilineTitle, [NSString stringWithFormat:@"%@\n", localBranchName], multilineTitleAttributes);
+      [multilineTitle addAttribute:NSFontAttributeName value:(id)boldFont range:localBranchNameRange];
+      [multilineTitle addAttribute:NSForegroundColorAttributeName value:darkColor range:localBranchNameRange];
+      
+      // Upstream may be local or remote, so we look for slash to make the branch name more emphasized
+      NSRange upstreamBranchNameRange = NSMakeRange(multilineTitle.length + 2, upstreamName.length); // +2 to exclude "⬅ "
+      NSRange slashRange = [upstreamName rangeOfString:@"/"];
```

Comment:
Use the API here, don't search for `/`. The fact the API uses `-rangeOfString:` internally is an implementation detail. What matters is that we don't have code in the app that duplicates functionality of the API. It's increasing the risk of bugs and pretty much guarantees that improving the API later on or changing its implementation will get things out of sync.